loadtest/control/gencontroller/actions.go

```diff
@@ -368,3 +411,33 @@ func (c *GenController) joinTeam(u user.User) control.UserActionResponse {
 
 	return control.UserActionResponse{Info: fmt.Sprintf("joined team %s", team.Id)}
 }
+
+// chooseChannel will pick a channelID randomly from the range of indexes.
+// If the chosen channelID has exceeded the number of channelmembers, it will
+// select another one in the range until it has found one.
+func chooseChannel(dist ChannelsDistribution, u user.User) (string, error) {
+	minIndex := int(dist.MinIndexRange * float64(len(st.channels)))
+	maxIndex := int(dist.MaxIndexRange * float64(len(st.channels)))
+
+	if maxIndex-minIndex <= 1 {
+		return "", errors.New("not enough channels to select from; either increase range or increase number of channels to create")
+	}
+
+	var channelID string
+	for {
+		target := rand.Intn(maxIndex-minIndex) + minIndex
+		if target >= maxIndex {
+			target = maxIndex - 1
+		}
```

Comment:
Why do we need the `target >= maxIndex` check here? Isn't `target` always in the `[minIndex, maxIndex)` interval?

Reply:
We could check against len(st.channels) to be extra sure that we don't access memory outside of the array bounds, though.

Reply:
Yeah, it depends on how you configure your channel range intervals in the config. If you keep the minIndex of the next range higher than maxIndex of the previous range, then it should be fine. But if the user makes a mistake, this is more of a safety net.

> We could check against len(st.channels) to be extra sure that we don't access memory outside of the array bounds, though.

Right, so the highest possible `maxIndex` is basically `len(st.channels)`. I am essentially ensuring that the `target` is up to `len(st.channels)-1`.

LMK if you prefer to change it to len(st.channels)

Reply:
Ah, that's true! I guess that could be validated, but I also don't think it's super important. Thanks for the discussion here!